libPhoneNumberTests/metadataGenerator.swift

```
@@ -0,0 +1,125 @@
+  }
+  dataTask.resume()
+
+  semaphore.wait()
+
+  if let error = resultError {
+    throw error
+  }
+
+  if let data = resultData {
+    guard let string = String(data: data, encoding: .utf8) else {
+      throw GeneratorError.dataNotString
+    }
+
+    return string
+  }
+
+  return nil
+}
+
+func loadJS(from url: URL, to context: JSContext) {
+  guard let script = try? synchronouslyLoadStringResource(from: url) else {
+    fputs("Cannot load dependency at \(url)", __stderrp)
+    exit(1)
+  }
+
+  context.evaluateScript(script)
+}
+
+// Create JavaScript context.
+let context = JSContext()!
+
+// Load required dependencies.
+let googleClosure = URL(
+  string: "http://cdn.rawgit.com/google/closure-library/master/closure/goog/base.js")!
+loadJS(from: googleClosure, to: context)
+
+let jQuery = URL(string: "http://code.jquery.com/jquery-1.8.3.min.js")!
+loadJS(from: jQuery, to: context)
+
+// Evaluate requires.
+let requires = """
+  goog.require('goog.proto2.Message');
+  goog.require('goog.dom');
+  goog.require('goog.json');
+  goog.require('goog.array');
+  goog.require('goog.proto2.ObjectSerializer');
+  goog.require('goog.string.StringBuffer');
+  goog.require('i18n.phonenumbers.metadata');
+  """
+context.evaluateScript(requires)
+
+// Load metadata file from GitHub.
+let phoneMetadata = "https://raw.githubusercontent.com/googlei18n/libphonenumber/master/javascript/i18n/phonenumbers/metadata.js"
+let phoneMetadataForTesting = "https://raw.githubusercontent.com/googlei18n/libphonenumber/master/javascript/i18n/phonenumbers/metadatafortesting.js"
+let shortNumberMetadata = "https://raw.githubusercontent.com/googlei18n/libphonenumber/master/javascript/i18n/phonenumbers/shortnumbermetadata.js"
+
+let currentDir = FileManager.default.currentDirectoryPath
+let baseURL = URL(fileURLWithPath: currentDir).appendingPathComponent("generatedJSON")
+
+// Phone metadata.
+if let metadata = try? synchronouslyLoadStringResource(from: URL(string: phoneMetadata)!) {
```

If this fails isn't it going to fail silently? This is probably not something we want to fail silently.

---

Added the catch block, and terminate the script if failed